test-app/runtime/src/main/cpp/console/Console.cpp

```diff
@@ -0,0 +1,332 @@
+#include <include/v8.h>
+#include <assert.h>
+#include <android/log.h>
+#include <cstdlib>
+#include <chrono>
+#include <iomanip>
+#include <sstream>
+#include <V8GlobalHelpers.h>
+#include <v8_inspector/src/inspector/v8-log-agent-impl.h>
+#include <JsV8InspectorClient.h>
+#include "Console.h"
+
+namespace tns {
+
+v8::Local<v8::Object> Console::createConsole(v8::Local<v8::Context> context, const std::string& filesPath) {
+    v8::Context::Scope contextScope(context);
+    v8::Isolate* isolate = context->GetIsolate();
+
+    v8::Local<v8::Object> console = v8::Object::New(isolate);
+    bool success = console->SetPrototype(context, v8::Object::New(isolate)).FromMaybe(false);
+
+    assert(success);
+
+    std::map<std::string, double> timersMap;
+    Console::s_isolateToConsoleTimersMap.insert(std::make_pair(context->GetIsolate(), timersMap));
+
+    bindFunctionProperty(context, console, "assert", assertCallback);
+    bindFunctionProperty(context, console, "error", errorCallback);
+    bindFunctionProperty(context, console, "info", infoCallback);
+    bindFunctionProperty(context, console, "log", logCallback);
+    bindFunctionProperty(context, console, "warn", warnCallback);
+    bindFunctionProperty(context, console, "dir", dirCallback);
+    bindFunctionProperty(context, console, "trace", traceCallback);
+    bindFunctionProperty(context, console, "time", timeCallback);
+    bindFunctionProperty(context, console, "timeEnd", timeEndCallback);
+
+    return console;
+}
+
+void Console::sendToADBLogcat(const std::string& message, android_LogPriority logPriority) {
+    // split strings into chunks of 4000 characters
+    // __android_log_write can't send more than 4000 to the stdout at a time
+
+    auto messageLength = message.length();
+    int maxStringLength = 4000;
+
+    if (messageLength < maxStringLength) {
+        __android_log_write(logPriority, Console::LOG_TAG, message.c_str());
+    } else {
+        for (int i = 0; i < messageLength; i += maxStringLength) {
+            auto messagePart = message.substr(i, maxStringLength);
+
+            __android_log_write(logPriority, Console::LOG_TAG, messagePart.c_str());
+        }
+    }
+}
+
+void Console::sendToDevToolsFrontEnd(v8::Isolate* isolate, const std::string& message, const std::string& logLevel) {
+    if (!JsV8InspectorClient::inspectorIsConnected()) {
+        return;
+    }
+
+    auto stack = v8::StackTrace::CurrentStackTrace(isolate, 1, v8::StackTrace::StackTraceOptions::kDetailed);
+
+    auto frame = stack->GetFrame(0);
+
+    // will be no-op in non-debuggable builds
+    v8_inspector::V8LogAgentImpl::EntryAdded(message, logLevel, ArgConverter::ConvertToString(frame->GetScriptNameOrSourceURL()), frame->GetLineNumber());
+}
+
+const std::string& buildLogString(const v8::FunctionCallbackInfo<v8::Value>& info) {
+    auto isolate = info.GetIsolate();
+
+    v8::HandleScope scope(isolate);
+
+    std::stringstream ss;
+
+    auto argLen = info.Length();
+    if (argLen) {
```

Comment:
suggestion: if no arguments are passed maybe we can print a new line, since that's the common behavior

Reply:
Nice catch! That will align the behavior with how node's `console.log()` with empty args does.